keps/sig-network/2433-topology-aware-hints/README.md

```diff
@@ -369,12 +376,27 @@ of the following scenarios:
 2. A new Node results in a Service that is able to achieve an endpoint
    distribution below 20% for the first time.
 
+### SameZone Heuristic
+The simplest of all heuristics, this will simply copy the zone of an endpoint to
+the hint of the same endpoint. If there are not endpoints available within a zone,
+that will mean no endpoints will have hints for that zone. As described above,
+when Kube-Proxy cannot find any endpoints with hints for its zone, it will route
```

Comment:
this implies prescribing changes on the proxies implementations ...

Reply:
Yep, proxies would have to remove the check for the annotation, but otherwise, I don't think this would require any additional changes.

Reply:
What I understand from the kube-proxy section above is that if there is a service that has any endpoint (regardless of zone) without a hint - kube proxy on all zones will ignore hints for this service -  `if any endpoints for a Service do not have a hint, kube-proxy will ignore all hints` 

While the SameZone section basically say that when Kube-Proxy cannot find any endpoints with hints **for its zone** it will ignore the hints and will route the traffic evenly. (and kube-proxies on other zones will not ignore the hints?)

Reply:
While proxies would need to change, it's probably a good, safe-fallback to assume "there are no endpoints for my zone, I doubt you meant there to be no endpoints at all".  I think?

Reply:
or just configure the proxy with the 3 different behaviors for hints
1. Strict: only do what hints says
2. Loose (default): absent of hints does not mean drop, do your best
3. None: what are hints? :)